scripts/eval/eval.py

```diff
@@ -160,123 +159,75 @@ def evaluate_model(
         torch.cuda.synchronize()
     b = time.time()
 
-    log.info(f'Ran {model_cfg.model_name} eval in: {b-a} seconds')
+    log.info(f'Ran {model_name} eval in: {b-a} seconds')
     return (trainer, logger_keys, eval_gauntlet_callback, eval_gauntlet_df)
 
 
 def main(cfg: DictConfig) -> Tuple[List[Trainer], pd.DataFrame]:
+    cfgs: Tuple[Dict[str, Any], EvalConfig] = make_dataclass_and_log_config(
+        cfg, EvalConfig, EVAL_CONFIG_KEYS, icl_tasks_required=True)
+    logged_cfg, eval_config = cfgs
+
     # Run user provided code if specified
-    code_paths = pop_config(cfg,
-                            'code_paths',
-                            must_exist=False,
-                            default_value=[],
-                            convert=True)
-    for code_path in code_paths:
+    for code_path in (eval_config.code_paths or []):
```

Comment:
This should be the first thing that runs. So even though you won't have a structured config yet, this should be above the parsing